static-build/qemu-virtiofs/Dockerfile

```diff
@@ -19,37 +19,56 @@ RUN apt-get --no-install-recommends install -y \
 	    flex \
 	    gawk \
 	    libaudit-dev \
+	    libblkid-dev \
 	    libcap-dev \
 	    libcap-ng-dev \
 	    libdw-dev \
 	    libelf-dev \
+	    libffi-dev \
 	    libglib2.0-0 \
 	    libglib2.0-dev \
 	    libglib2.0-dev git \
 	    libltdl-dev \
+	    libmount-dev \
 	    libpixman-1-dev \
 	    libpmem-dev \
 	    libseccomp-dev \
+	    libseccomp2 \
+	    libselinux1-dev \
 	    libtool \
-	    patch \
+	    make \
 	    pkg-config \
 	    pkg-config \
 	    python \
 	    python-dev \
 	    rsync \
 	    seccomp \
-	    libseccomp2 \
 	    zlib1g-dev
 
 RUN cd  .. && git clone "${QEMU_VIRTIOFS_REPO}" qemu-virtiofs
 RUN git checkout "${QEMU_VIRTIOFS_TAG}"
-ADD qemu/patches/virtiofsd/0001-add-time-to-seccomp.patch /root/0001-add-time-to-seccomp.patch
-ADD qemu/patches/virtiofsd/0002-libvhost-user-Fix-the-VHOST_USER_PROTOCOL_F_SLAVE_SE.patch /root/0002-libvhost-user-Fix-the-VHOST_USER_PROTOCOL_F_SLAVE_SE.patch
-RUN patch -p1 < /root/0001-add-time-to-seccomp.patch
-RUN patch -p1 < /root/0002-libvhost-user-Fix-the-VHOST_USER_PROTOCOL_F_SLAVE_SE.patch
+
 ADD scripts/configure-hypervisor.sh /root/configure-hypervisor.sh
-RUN PREFIX="${PREFIX}" /root/configure-hypervisor.sh -s kata-qemu | sed -e 's|--enable-rbd||g' -e 's|--disable-seccomp||g' | xargs ./configure \
+ADD qemu/patches/ /root/kata_qemu_patches
+# Apply virtiofs specific patches
+RUN  \
+	for patch in $(find /root/kata_qemu_patches/virtiofsd/${QEMU_VIRTIOFS_TAG}/ -name '*.patch'); do\
```

Comment:
This doesn't look reliable - I think we need to sort the patches before applying as on https://github.com/kata-containers/packaging/pull/1005.

Reply:
Agree with @jodh-intel. We need a more robust approach here. 
@jcvenegas Can you explain how you are applying the patches and whats changed.

Reply:
Also -- I'm not sure if we need to carry any of the virtiofs patches, at least for 5.0.

Reply:
@jodh-intel done now is using sort to apply as you suggested
@amshinde @egernst   the patches are applied in the following order
1. First look and apply  patches specific to the tag for virtiofsd (this is in case we need to fix something before a tag is generated by virtios folks) : Today there is not additional patches
2. Look for patches specific to the qemu branch is based and apply them